src/pages/program/[slug]/mentors/index.tsx

```diff
@@ -24,20 +27,53 @@ const ViewMentorsPage: Page = () => {
       profileType: ProfileType.Mentor,
     },
   });
+  const { data: profileTagsData } = useGetProfileTagsByProgramQuery({
+    variables: { programId: currentProgram?.programId! },
+  });
   let unfiltered = data?.getProfiles || [];
+  const [mentors, setMentors] = useState(unfiltered);
+
+  useEffect(() => {
+    if (unfiltered) {
+      setMentors(unfiltered);
+    }
+  }, [unfiltered]);
 
   const fuse = new Fuse(unfiltered, {
     keys: ["user.firstName", "user.lastName", "profileJson"],
   });
-  const mentors = searchText
-    ? fuse.search(searchText).map((x) => x.item)
-    : unfiltered;
+
+  const filterMentors = (tags: string[]) => {
+    let newMentors = searchText
+      ? fuse.search(searchText).map((x) => x.item)
+      : unfiltered;
+    if (!tags.length) {
+      setMentors(newMentors);
+      return;
+    }
+    newMentors = newMentors.filter((profile) => {
+      const profileTagIds = profile.profileTags.map(
+        (profileTag) => profileTag.profileTagId
+      );
+      for (const tag of tags) {
+        if (!profileTagIds.includes(tag)) {
+          return false;
+        }
+      }
+      return true;
+    });
+    setMentors(newMentors);
+  };
+
+  useEffect(() => {
+    filterMentors(filteredTags);
```

Comment:
Making `filterMentors` not set state and instead be reliant on the other state variables would remove the need of this.